### DISCUSSION TITLE
sixel images render smaller than geomoetry param, hidpi issue?

### DISCUSSION THREAD

**Discussion by iambumblehead** [ImageMagick/ImageMagick#7168] [Help]
### ImageMagick version

Version: ImageMagick 7.1.1-29

### Operating system, version and so on

arch linux

### Description

It seems sixel images are rendered smaller than the declared geometry. Below is a screenshot of a terminal where the first command echoes the size of the terminal and the second command uses imagemagick to render a sixel image at the "full size" reported by the terminal.

**Problem: the image is about 3 times smaller than it should be.** Larger geometry values produce larger images that look pixelated.

<img src="https://github.com/ImageMagick/ImageMagick/assets/2058705/0451a76e-4805-4557-ad78-f2c8ba3b3b78" width="700px" />

-----------------------------------

This machine (dell xps13) uses a hidpi screen "scaled" to 300% in Gnome and, if this is related, would you recommend the best way to handle this programmatically through a script intended for any device?

<img src="https://github.com/ImageMagick/ImageMagick/assets/2058705/f0975b20-328b-4738-b4a1-3b4d534ad36e" width="600px" />

------------------------------------
related,
 * guix artwork with svg and png files, `git clone https://git.savannah.gnu.org/git/guix/guix-artwork.git/`
 * xterm/foot escape sequence command `IFS=";" read -d t -sa REPLY -t 2 -p $'\e[14;2t' >&2; echo "${REPLY[2]} ${REPLY[1]}"`

**Comment by snibgo** [comment]
As you have IM v7, I suggest you use `magick` instead of `convert`.

`-geometry` has no effect in your command. Use `-size` to influence the size of the output image, like this:
```
magick -size 700x474 Guix_logo.svg sixel:-
```

**Reply by iambumblehead** [reply]
thanks for the suggestion, however it did not resolve the issue,

<img src="https://github.com/ImageMagick/ImageMagick/assets/2058705/a10a9b1b-b2d1-4860-a1c5-8e0f6cfe6790" width="400px" />

**Comment by snibgo** [comment]
If you want to use `-geometry`, put it after reading the SVG, not before.

**Reply by iambumblehead** [reply]
<img src="https://github.com/ImageMagick/ImageMagick/assets/2058705/c6a170c6-1737-4a2f-9533-081f49cbfbde" width="400px" />

**Comment by snibgo** [comment]
You are rasterizing an SVG file, which probably contains vector data. Try using `-density`, such as:
```
magick -density 1200 Guix_logo.svg out.png
```

**Reply by iambumblehead** [reply]
<img src="https://github.com/ImageMagick/ImageMagick/assets/2058705/00c1fa8f-ee2d-4df2-9720-842711ae5f5c" width="400px" />

top message is updated to include reference to the repo with the svg and png image

**Reply by iambumblehead** [reply]
related on-going messages here https://codeberg.org/dnkl/foot/issues/1643

**Comment by iambumblehead** [comment]
The solution will come from the terminal emulator foot. Currently using xterm gives the correct behaviour
